ratpack-core/src/main/java/ratpack/health/HealthCheckHandler.java

```diff
@@ -0,0 +1,435 @@
+   */
+  public HealthCheckHandler(String healthCheckName) {
+    this(healthCheckName, DEFAULT_CONCURRENCY_LEVEL);
+  }
+
+  /**
+   * Execute all registered health check with the given {@code concurrencyLevel}.
+   * @param concurrencyLevel the level of parallelism for health check promises execution. Define number of health checks (as promises) to be run in parallel.
+   * <ul>
+   *   <li><strong>0</strong> - infinite potential parallelism - up to the event loop number of threads</li>
+   *   <li><strong>1</strong> - serial, promises run one by one in sequence</li>
+   *   <li><strong>2</strong> - max 2 at a time promises run in parallel</li>
+   *   <li><strong>3</strong> - max 3 at a time promises run in parallel</li>
+   *   <li><strong>n</strong> - max n at a time promises run in parallel</li>
+   * </ul>
+   */
+  public HealthCheckHandler(int concurrencyLevel) {
+    this(DEFAULT_NAME_TOKEN, concurrencyLevel);
+  }
+
+  /**
+   * Execute health check of the given name (if different than {@link ratpack.health.HealthCheckHandler#DEFAULT_NAME_TOKEN} and
+   * given {@code concurrencyLevel}.
+   *
+   * @param healthCheckName the name of health check
+   * @param concurrencyLevel the level of parallelism for health check promises execution. Define number of health checks (as promises) to be run in parallel.
+   * <ul>
+   *   <li><strong>0</strong> - infinite potential parallelism - up to the event loop number of threads</li>
+   *   <li><strong>1</strong> - serial, promises run one by one in sequence</li>
+   *   <li><strong>2</strong> - max 2 at a time promises run in parallel</li>
+   *   <li><strong>3</strong> - max 3 at a time promises run in parallel</li>
+   *   <li><strong>n</strong> - max n at a time promises run in parallel</li>
+   * </ul>
+   */
+  protected HealthCheckHandler(String healthCheckName, int concurrencyLevel) {
+    this.name = healthCheckName;
+    this.concurrencyLevel = concurrencyLevel;
+  }
+
+  /**
+   * Get the level of parallelism in non-blocking health check execution.
+   *
+   * @return concurrency level as defined in {@link ratpack.health.HealthCheckHandler#HealthCheckHandler} constructor. Parameter {@code concurrencyLevel}.
+   */
+  public int getConcurrencyLevel() {
+    return concurrencyLevel;
+  }
+
+  /**
+   * Get assigned health check name. If defined health check of this name is executed.
+   *
+   * @return health check name to be executed
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Run individual health check with the given name or all health checks.
+   * The {@code concurrencyLevel} determines parallelism in health check execution.
+   *
+   * @param context request context and exec control as well
+   * @throws Exception if anything goes wrong, exception will be implicitly passed to the context's {@link Context#error(Throwable)} method
+   */
+  @Override
+  public void handle(Context context) throws Exception {
+    if (!name.equals(DEFAULT_NAME_TOKEN)) {
```

Comment:
This logic doesn't seem right.

What we want to support is running a particular check if its name was specified as a path token. This classes param should indicate the name of the path token to look for. If the token is present, we only render the check whose name matches.